All_home_works/hw6/task2.py

```diff
@@ -101,20 +102,20 @@ def reset_results(homework=None):
         if homework is None:
             Teacher.homework_done = defaultdict(set)
         else:
-            del Teacher.homework_done[homework]
+            Teacher.homework_done.pop(homework)
 
     @staticmethod
     def create_homework(text, deadline):
         return Homework(text=text, deadline=deadline)
 
 
-class InstanceNotHomeworkError(BaseException):
+class InstanceNotHomeworkError(TypeError):
     def __init__(self, homework):
         self.messages = f"You gave a not Homework object, Got {type(homework)}"
```

Comment:
`a not` -> `not a`


You can use some samples from the Python itself for error messages:  try `int([])`

Reply:
Like this?
`raise InstanceNotHomeworkError('You gave a not Homework object, Got', type(homework))`

Reply:
I mean that your message could use a style that is used by Python error messages.